src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/IntentionalRerouteSpec.groovy

```diff
@@ -1,57 +1,62 @@
+import static org.openkilda.testing.Constants.DEFAULT_COST
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+import static spock.util.matcher.HamcrestSupport.expect
+
 import org.openkilda.functionaltests.HealthCheckSpecification
 import org.openkilda.functionaltests.extension.tags.Tags
 import org.openkilda.functionaltests.helpers.PathHelper
 import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.functionaltests.helpers.factory.FlowFactory
+import org.openkilda.functionaltests.helpers.model.FlowEncapsulationType
 import org.openkilda.messaging.info.event.PathNode
 import org.openkilda.messaging.payload.flow.FlowState
-import org.openkilda.model.FlowEncapsulationType
 import org.openkilda.testing.service.traffexam.TraffExamService
-import org.openkilda.testing.tools.FlowTrafficExamBuilder
+
 import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Narrative
 import spock.lang.See
 import spock.lang.Shared
 
 import javax.inject.Provider
 
-import static com.shazam.shazamcrest.matcher.Matchers.sameBeanAs
-import static groovyx.gpars.GParsPool.withPool
-import static org.junit.jupiter.api.Assumptions.assumeTrue
-import static org.openkilda.functionaltests.extension.tags.Tag.HARDWARE
-import static org.openkilda.functionaltests.extension.tags.Tag.ISL_PROPS_DB_RESET
-import static org.openkilda.functionaltests.extension.tags.Tag.LOW_PRIORITY
-import static org.openkilda.testing.Constants.DEFAULT_COST
-import static org.openkilda.testing.Constants.WAIT_OFFSET
-import static spock.util.matcher.HamcrestSupport.expect
-
 @See("https://github.com/telstra/open-kilda/tree/develop/docs/design/hub-and-spoke/reroute")
 @Narrative("Verify that on-demand reroute operations are performed accurately.")
 
 class IntentionalRerouteSpec extends HealthCheckSpecification {
 
-    @Autowired @Shared
+    @Autowired
+    @Shared
     Provider<TraffExamService> traffExamProvider
+    @Autowired
+    @Shared
+    FlowFactory flowFactory
 
     @Tags(ISL_PROPS_DB_RESET)
     def "Not able to reroute to a path with not enough bandwidth available"() {
         given: "A flow with alternate paths available"
         def switchPair = switchPairs.all().neighbouring().withAtLeastNPaths(2).random()
-        def flow = flowHelperV2.randomFlow(switchPair)
-        flow.maximumBandwidth = 10000
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(switchPair)
+                .withBandwidth(10000)
+                .build().create()
 
-        def currentPathDto = northbound.getFlowPath(flow.flowId)
-        def currentPath = PathHelper.convert(currentPathDto)
-        def currentPathNodesV2 = PathHelper.convertToNodesV2(currentPathDto)
+        def currentPathEntities = flow.retrieveAllEntityPaths()
+        def currentPath = currentPathEntities.getPathNodes()
+        def currentPathNodesV2 = PathHelper.convertToNodesV2(currentPath)
```

Comment:
can be replaced: `currentPathEntities.flowPath.path.forward.nodes.nodes`

Reply:
Done